tensorflow/core/kernels/mkl_fused_batch_norm_op.cc

```diff
@@ -143,83 +179,152 @@ class MklFusedBatchNormFwdPrimitive : public MklPrimitive {
+#endif
 
-    // fwd desc & primitive desc
+    // Create forward BatchNorm descriptor and primitive descriptor.
+#ifdef ENABLE_MKLDNN_V1
+    auto fwd_desc = batch_normalization_forward::desc(
+        context_.pkind, src_md, fwdParams.eps,
+        (mkldnn::batch_normalization_flags), context_.flags);
+#else
     auto fwd_desc = batch_normalization_forward::desc(
         context_.pkind, src_md, fwdParams.eps, context_.flags);
-    auto fwd_pd =
-        batch_normalization_forward::primitive_desc(fwd_desc, cpu_engine_);
-
-    // memory primitive
-    context_.src_mem.reset(new memory({src_md, cpu_engine_}, DummyData));
-    context_.dst_mem.reset(new memory(fwd_pd.dst_primitive_desc(), DummyData));
-
-    if (context_.flags & use_scale_shift) {
-      auto weights_desc = memory::desc({2, fwdParams.depth}, MklDnnType<U>(),
-                                       memory::format::nc);
-      context_.weights_mem.reset(
-          new memory({weights_desc, cpu_engine_}, DummyData));
+#endif
+
+    context_.fwd_pd.reset(new BatchNormFwdPd(fwd_desc, cpu_engine_));
+
+    // Create memory primitive based on dummy data
+    context_.src_mem.reset(new MEMORY_CONSTRUCTOR(
+        context_.fwd_pd->PRIMITIVE_DESC_SRC, cpu_engine_, DummyData));
+    context_.dst_mem.reset(new MEMORY_CONSTRUCTOR(
+        context_.fwd_pd->PRIMITIVE_DESC_DST, cpu_engine_, DummyData));
+
+    memory::dims s_dims = {2, fwdParams.depth};
+    memory::dims m_dims = {1, fwdParams.depth};
+    if (context_.flags & (int)BN_FLAGS::use_scale_shift) {
+      context_.weights_mem.reset(new MEMORY_CONSTRUCTOR_USING_MEM_PD(
+          s_dims, U, MEMORY_FORMAT::nc, cpu_engine_, DummyData));
     }
 
-    if (fwdParams.training || (context_.flags & use_global_stats)) {
-      auto mean_desc = memory::desc({1, fwdParams.depth}, MklDnnType<U>(),
-                                    memory::format::nc);
-      context_.mean_mem.reset(new memory({mean_desc, cpu_engine_}, DummyData));
+    if (fwdParams.training ||
+        (context_.flags & (int)BN_FLAGS::use_global_stats)) {
+      context_.mean_mem.reset(new MEMORY_CONSTRUCTOR_USING_MEM_PD(
+          m_dims, U, MEMORY_FORMAT::nc, cpu_engine_, DummyData));
 
-      auto variance_desc =
-          memory::desc({1, fwdParams.depth}, MklDnnType<U>(), memory::nc);
-      context_.variance_mem.reset(
-          new memory({variance_desc, cpu_engine_}, DummyData));
+      context_.variance_mem.reset(new MEMORY_CONSTRUCTOR_USING_MEM_PD(
+          m_dims, U, MEMORY_FORMAT::nc, cpu_engine_, DummyData));
     }
 
-    // BatchNorm forward primitive
-    if (!fwdParams.training && !(context_.flags & use_global_stats)) {
-      if ((context_.flags & use_scale_shift) && mkldnn_use_scaleshift) {
+    // BatchNorm forward primitive.
+    if (!fwdParams.training &&
+        !(context_.flags & (int)BN_FLAGS::use_global_stats)) {
+#ifdef ENABLE_MKLDNN_V1
+      if ((context_.flags & (int)BN_FLAGS::use_scale_shift) &&
+          mkldnn_use_scaleshift) {
```

Comment:
`BN_FLAGS::use_scale_shift` is set to `mkldnn_use_scaleshift` in both v0 and v1. Why do we need `mkldnn_use_scaleshift` here? Also, I believe the `&&` in front of `mkldnn_use_scaleshift` is a typo and should be `&` (should `mkldnn_use_scaleshift` remain here).